### PR TITLE
Move the SDK out of opentelemetry crate, which is now the API only

### LINKED ISSUE #1186: What is the difference between the API and SDK?
As a user of `opentelemetry`, there are quite a few terms to learn when navigating the various packages to achieve some goal. 

One of the things I've found confusing is the separation between the `opentelemetry_api` and `opentelemetry_sdk`. As an outsider that may be missing some project-specific context, those two terms sound like they ought to be interchangeable. My general understanding of SDKs are that they're libraries or sets of libraries that help you build against or on top of APIs. I don't understand why they're separate in OpenTelemetry world and it always feels somewhat random to me what things I need to import from the API or SDK crate.

The OpenTelemetry [glossary](https://opentelemetry.io/docs/concepts/glossary/) doesn't help that much:

> Application Programming Interface. In the OpenTelemetry project, used to define how telemetry data is generated per [Data Source](https://opentelemetry.io/docs/concepts/glossary/#data-source).

> Short for Software Development Kit. Refers to a telemetry SDK that denotes a [Library](https://opentelemetry.io/docs/concepts/glossary/#library) that implement the OpenTelemetry [API](https://opentelemetry.io/docs/concepts/glossary/#api).

Could this be clarified further in the docs or (I assume not) could the crates just be combined?

### PR DESCRIPTION
**Pull request by shaun-cox**
Fixes #1186

## Changes

- remove the re-exports of `opentelemetry_sdk` types from `opentelemetry`
- fix all places that were using `opentelemetry::sdk` to use `opentelemetry_sdk`
- removed features of `opentelemetry` that did not pass-thru to `opentelemetry_api`

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by djc** [comment]
At this point, should we just move all of the opentelemetry_api code back into the opentelemetry directory? Seems simpler to me. (If there is some reason to defer that to a separate PR, that's fine too, but it seems like it could belong here.)

**Comment by shaun-cox** [comment]
> At this point, should we just move all of the opentelemetry_api code back into the opentelemetry directory?

My vote would be yes, as otherwise there are two choices for which crate to depend on for the api.  I would like to leave it for a follow-up PR though.

**Comment by shaun-cox** [comment]
> Just bump to 1.63, should probably be in a separate PR though.

I can do this, but would like clarification before I start.  (@jtescher, would like your input too.)
There are over 50 places mentioning "1.60".  Would you like them all updated to "1.63" (sans the history in CHANGLELOG.md references), or just the crates that depend on tokio that need 1.63?  Thanks!

**Comment by TommyCpp** [comment]
Yeah we usually just change them all to avoid any gap between crates.

**Comment by jtescher** [comment]
yeah 1.63 seems fine, eventually having api and sdk drift in terms of msrv would be good but simplicity and correctness are probably more important as we head toward 1.0

**Comment by shaun-cox** [comment]
Waiting for https://github.com/open-telemetry/opentelemetry-rust/pull/1203 to be merged, then I'll rebase and squash these commits on that.

File: opentelemetry/CHANGELOG.md
Comment (cijothomas) [inline opentelemetry/CHANGELOG.md:8]: add a note saying "opentelemetry_sdk" crate must be referenced instead, if relying on SDK features.

**Review by lzchen — APPROVED** [review]

**Review by hdost — APPROVED** [review]
As mentioned during the SIG we'll disable build of z pages for now

**Review by lalitb — APPROVED** [review]

File: Cargo.toml
Comment (cijothomas) [reply Cargo.toml:25]: https://github.com/open-telemetry/opentelemetry-rust/issues/1208 opened an issue to track the full removal. It was generally agreed in the SIG call to disable it in this PR to unblock this PR and get this merged.

Comment (hdost) [reply Cargo.toml:25]: We were attempting to unlink this PR from being tied to this whole discussion about zpages by someone only disabling build. 
So that it wouldn't block this PR. 
I guess let's discuss the fate in #1208

**Comment by cijothomas** [comment]
@TommyCpp to merge this PR, and we can work on actual deprecation as a separate PR, so that this is unblocked.

**Comment by shaun-cox** [comment]
@TommyCpp, thanks for the zpages fix. I've rebased this and now it all works with zpages included. This PR is ready for final review and merge if everyone is okay with it.

**Review by TommyCpp — APPROVED** [review]

**Comment by TommyCpp** [comment]
Thanks for working on this!

**Pull request merged**
